src/deep_learning_container.py

```diff
@@ -44,21 +102,25 @@ def _retrieve_instance_id():
     Retrieve instance ID from instance metadata service
     """
     instance_id = None
-    # We can't use IMDSv2 here which needs token, as adding it mandates docker container to add additional parameter "--network host", else it hangs.
-    url = "http://169.254.169.254/latest/meta-data/instance-id"
-    response = requests_helper(url, timeout=0.1)
+    instance_url = "http://169.254.169.254/latest/meta-data/instance-id"
+    
+    instance_id = requests_helper_imds(instance_url)
 
-    if response is not None and not (400 <= response.status_code < 600):
-        instance_id = _validate_instance_id(response.text)
+    if instance_id is None:
```

Comment:
do you need to explicitly check for "None" or is empty string or 0 allowed? Those cases will be missed in the two if statements.

Additionally, why are these two separate if statements instead of if/else?